src/input/isf.c

```diff
@@ -0,0 +1,529 @@
+}
+
+static gboolean has_header(GString *buf)
+{
+    return buf->len >= MIN_INITIAL_OFFSET;
+}
+
+static char *find_data_section(GString *buf)
+{
+    const char *curve = "CURVE #";
+
+    char *data_ptr;
+    size_t offset, data_length;
+
+    data_ptr = g_strstr_len(buf->str, buf->len, curve);
+    if(data_ptr == NULL)
+        return NULL;
+
+    data_ptr += strlen(curve);
+    offset = data_ptr - buf->str;
+    if(offset >= buf->len)
+        return NULL;
+
+    data_length = (size_t) *data_ptr - 48;
+    data_ptr += 1 + data_length;
+    offset = data_ptr - buf->str;
+
+    if(offset >= buf->len)
+        return NULL;
+
+    return data_ptr;
+}
+
+static void extract_channel_name(struct context *inc, const char *beg)
+{
+    size_t i, channel_ix;
+
+    channel_ix = 0;
+    /* Isf wfid looks something like WFID "Ch1, ..."; thus we must skip character '"' */
+    i = 1;
+    while(beg[i] != ',' && beg[i] != '"' && channel_ix < MAX_CHANNEL_NAME_SIZE - 1)
+        inc->channel_name[channel_ix++] = beg[i++];
+}
+
+static void find_string_value(const char *beg, char *value, size_t value_len)
+{
+    size_t i, value_idx;
+
+    value_idx = 0;
+    i = 0;
+    while(beg[i] != ';' && value_idx < value_len)
+        value[value_idx++] = beg[i++];
+    value[value_idx] = 0;
+}
+
+static int find_encoding(const char *beg)
+{
+    char value[10];
+
+    find_string_value(beg, value, 9);
+
+    if(strcmp(value, "BINARY") != 0){
+        sr_err("Only binary encoding supported.");
+        return SR_ERR_NA;
+    }
+
+    return SR_OK;
+}
+
+static int find_waveform_type(struct context *inc, const char *beg)
+{
+    char value[10];
+
+    find_string_value(beg, value, 9);
+
+    if(strcmp(value, "ANALOG") == 0)
+        inc->wfmtype = ANALOG;
+    else if(strcmp(value, "RF_FD") == 0)
+        inc->wfmtype = RF_FD;
+    else
+        return SR_ERR_DATA;
+
+    return SR_OK;
+}
+
+static int process_header_item(const char *beg, struct context *inc, enum header_items_enum item)
+{
+    int ret;
+
+    switch (item) {
+        case YOFF:
+            inc->yoff = (float) g_ascii_strtod(beg, NULL);
+            break;
+        case YZERO:
+            inc->yzero = (float) g_ascii_strtod(beg, NULL);
+            break;
+        case YMULT:
+            inc->ymult = (float) g_ascii_strtod(beg, NULL);
+            break;
+        case XINCR:
+            inc->xincr = (float) g_ascii_strtod(beg, NULL);
+            break;
+        case BYTNR:
+            inc->bytnr = (int) g_ascii_strtoll(beg, NULL, 10);
+            break;
+        case BYTE_ORDER:
+            if(strncmp(beg, "LSB", 3) == 0)
+                inc->byte_order = LSB;
+            else if(strncmp(beg, "MSB", 3) == 0)
+                inc->byte_order = MSB;
+            else
+                return SR_ERR_DATA;
+            
+            break;
+        case BN_FMT:
+            if(strncmp(beg, "RI", 2) == 0)
+                inc->bn_fmt = RI;
+            else if(strncmp(beg, "FP", 2) == 0)
```

Comment:
In [MSO4000 and DPO4000 Series Digital Phosphor Oscilloscopes Programmer Manual](https://download.tek.com/manual/077024801web.pdf) there are `RI` (signed integer) and `RP` (positive integer), see `WFMOutpre:BN_Fmt`.

Where did you get `FP` (floating point)?

Reply:
I used [this manual](https://www.manualslib.com/manual/1375808/Tektronix-Tds5000b-Series.html?page=816) which specifies `FP` (floating point).